### DISCUSSION TITLE
Problems with "Accept-Encoding: br" header

### DISCUSSION THREAD

**Discussion by phbender** [aws/aws-lambda-rust-runtime#575] [Q&A] [answered]
I have deployed a lambda bwhind API Gateway using `axum` etc., have been happy for a couple of minutes and now am completely lost.

This works:

* `curl -H "Accept-Encoding: gzip, deflate" -i https:/my-api`

This does not work:

* `curl -H "Accept-Encoding: gzip, deflate, br" -i https:/my-api`
* (note the `br` encoding here)

What I observe:

* The Gateway logs tell me that the handler has returned a `200` and a byte size which I totally expect (5 for `hello` response)
* The browser (or curl, ...) tells me `503`.

```
HTTP/2 503 
date: Tue, 06 Dec 2022 15:37:19 GMT
content-length: 35
apigw-requestid: cuwzYlQfoAMEJXQ=

{"message":"Internal Server Error"}%  
```

Does this sound familiar to someone here? The fact that the logs show the "correct" reponse tells me that the lambda either does not see or at least not act based on that header (indicator: the size). On the other hand I cannot imagine AWS having problems with `br` encoding.

**Comment by bnusunny** [comment]
Hello, this should work. Are you using REST API endpoint?

**Reply by phbender** [reply]
Hi bnsunny, I found the error, it has been very subtle. See answer below.

**Comment by phbender** [comment, marked as answer]
I specified the payload format version to be "1.0" instead of "2.0". I read about this parameter before but I would have expected it to explode with much more noise.

Double checked: by switching back and forth between both settings, I get the behaviour described above.

Bottom line: To deploy a Lambda handler using this crate, it can be deployed using code like this:

```terraform
resource "aws_apigatewayv2_api" "lambda" {
  name          = "some-gateway-name"
  protocol_type = "HTTP"
}

resource "aws_apigatewayv2_stage" "lambda" {
  api_id = aws_apigatewayv2_api.lambda.id

  name        = "v1"
  auto_deploy = true
}

resource "aws_apigatewayv2_integration" "lambda_app" {
  api_id = aws_apigatewayv2_api.lambda.id

  integration_uri    = aws_lambda_function.lambda_app.invoke_arn
  integration_type   = "AWS_PROXY"
  integration_method = "POST"
  payload_format_version = "2.0"
}
```

**Comment by bnusunny** [comment]
Yeah, for HTTP Endpoint, only payload format version 2.0 is supported.
